### ISSUE TITLE
[MxNet/ResNet50] Standard performance cannot be reproduced in resnet50

### ISSUE THREAD

**Issue by VectorSL** [NVIDIA/DeepLearningExamples#884]
As shown in [NVIDIA Data Center Deep Learning Product Performance](https://developer.nvidia.com/deep-learning-performance-training-inference) ，ResNet-50 v1.5 reached **2,751 images/sec** in A100 with the framework MxNet. But there isn't a specific container version，so I tried nvcr.io/nvidia/mxnet:20.12-py3 and nvcr.io/nvidia/mxnet:21.02-py3 from [NGC](https://ngc.nvidia.com/catalog/collections)，neither has lived up to expectations(about **1734 images/sec** ,  see details in 3). 
Here is my device info:
> A100-PCIE-40GB , NVIDIA-SMI 450.51.05    Driver Version: 450.51.05    CUDA Version: 11.0
Intel(R) Xeon(R) Platinum 8276L CPU @ 2.20GHz

Here's how I run it. Please correct me if there's any mistake, thx.

1. enter the image mxnet:20.12-py3
`nvidia-docker run --ipc=host --ulimit memlock=-1 --ulimit stack=67108864 --gpus all -it --rm -v /data2/mxnet_dataset/:/data/ nvcr.io/nvidia/mxnet:20.12-py3`

> ===========
== MXNet ==
===========
>NVIDIA Release 20.12 (build 18011711)
MXNet Version 1.8.0
Container image Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
Copyright (c) 2015-2020 by MXNet Contributors
Various files include modifications (c) NVIDIA CORPORATION.  All rights reserved.
NVIDIA modifications are covered by the license terms that apply to the underlying project or file.
NOTE: Legacy NVIDIA Driver detected.  Compatibility mode ENABLED.
NOTE: MOFED driver was detected, but nv_peer_mem driver was not detected.
Multi-node communication performance may be reduced.
root@a99c507d8a2a:/workspace#

2.  run the script
`/workspace/examples/resnet50v1.5#./runner  -n 1 -b 408 --data-root /data/mxnet_data/ --dtype float16`

3.  show the result

> [1,0]<stdout>:DLL 2021-03-24 07:42:46.785344 - PARAMETER arch : resnetv15  num_layers : 50  num_groups : 32  num_classes : 1000  batchnorm_eps : 1e-05  batchnorm_mom : 0.9  fuse_bn_relu : 1  fuse_bn_add_relu : 1  mode : train_val  seed : None  gpus : [0]  kv_store : horovod  dtype : float16  amp : False  batch_size : 408  num_epochs : 90  lr : 0.40800000000000003  lr_schedule : cosine  lr_factor : 0.256  lr_steps : []  warmup_epochs : 5  optimizer : sgd  mom : 0.875  wd : 3.0517578125e-05  label_smoothing : 0.1  mixup : 0  disp_batches : 20  model_prefix : model  save_frequency : -1  begin_epoch : 0  load : None  test_io : False  test_io_mode : train  log : log.log  dllogger_log : dllogger_log.log  workspace : ./  no_metrics : False  benchmark_iters : None  data_train : /data/mxnet_data/train.rec  data_train_idx : /data/mxnet_data/train.idx  data_val : /data/mxnet_data/val.rec  data_val_idx : /data/mxnet_data/val.idx  data_pred : None  data_backend : dali-gpu  image_shape : [4, 224, 224]  rgb_mean : [123.68, 116.779, 103.939]  rgb_std : [58.393, 57.12, 57.375]  input_layout : NCHW  conv_layout : NHWC  batchnorm_layout : NHWC  pooling_layout : NHWC  num_examples : 1281167  data_val_resize : 256  dali_separ_val : False  dali_threads : 3  dali_validation_threads : 10  dali_prefetch_queue : 2  dali_nvjpeg_memory_padding : 64  dali_fuse_decoder : 1  data_mxnet_threads : 40  random_crop : 0  random_mirror : 1  max_random_h : 0  max_random_s : 0  max_random_l : 0  min_random_aspect_ratio : 0.75  max_random_aspect_ratio : 1.33  max_random_rotate_angle : 0  max_random_shear_ratio : 0  max_random_scale : 1  min_random_scale : 1  max_random_area : 1  min_random_area : 0.05  min_crop_size : -1  max_crop_size : -1  brightness : 0  contrast : 0  saturation : 0  pca_noise : 0  random_resized_crop : 1
[1,0]<stderr>:[[1,0]<stderr>:07:42:46] ../src/storage/storage.cc:199: Using Pooled (Naive) StorageManager for CPU
[1,0]<stderr>:[07:42:56] ../src/storage/storage.cc:199: Using Pooled (Naive) StorageManager for GPU
[1,0]<stderr>:/workspace/examples/resnet50v1.5/dali.py:65: DeprecationWarning: Argument 'output_dtype' for operator 'CropMirrorNormalize' is now deprecated. Use 'dtype' instead.
[1,0]<stderr>:  self.cmnp = ops.CropMirrorNormalize(device="gpu",
[1,0]<stderr>:/workspace/examples/resnet50v1.5/dali.py:65: DeprecationWarning: Argument 'image_type' for operator 'CropMirrorNormalize' is now deprecated. The argument is no longer used and should be removed.
[1,0]<stderr>:  self.cmnp = ops.CropMirrorNormalize(device="gpu",
[1,0]<stderr>:/workspace/examples/resnet50v1.5/dali.py:100: DeprecationWarning: Argument 'output_dtype' for operator 'CropMirrorNormalize' is now deprecated. Use 'dtype' instead.
[1,0]<stderr>:  self.cmnp = ops.CropMirrorNormalize(device="gpu",
[1,0]<stderr>:/workspace/examples/resnet50v1.5/dali.py:100: DeprecationWarning: Argument 'image_type' for operator 'CropMirrorNormalize' is now deprecated. The argument is no longer used and should be removed.
[1,0]<stderr>:  self.cmnp = ops.CropMirrorNormalize(device="gpu",
[1,0]<stderr>:/workspace/examples/resnet50v1.5/dali.py:176: UserWarning: 1281167 training examples will be used, although full training set contains 1281231 examples
[1,0]<stderr>:  warnings.warn("{} training examples will be used, although full training set contains {} examples".format(args.num_examples, trainpipes[0].epoch_size("Reader")))
[1,0]<stderr>:/usr/local/lib/python3.8/dist-packages/nvidia/dali/plugin/base_iterator.py:156: Warning: Please set `reader_name` and don't set last_batch_padded and size manually  whenever possible. This may lead, in some situations, to miss some  samples or return duplicated ones. Check the Sharding section of the documentation for more details.
[1,0]<stderr>:  _iterator_deprecation_warning()
[1,0]<stderr>:/usr/local/lib/python3.8/dist-packages/nvidia/dali/plugin/mxnet.py:84: Warning: Please do not use `fill_last_batch` and use `last_batch_policy` instead.
[1,0]<stderr>:  _DaliBaseIterator.__init__(self, pipelines, size, reader_name, auto_reset,
[1,0]<stderr>:2021-03-24 07:43:01,767:WARNING: DALI iterator does not support resetting while epoch is not finished. Ignoring...
[1,0]<stderr>:2021-03-24 07:43:01,767:INFO: Starting epoch 0
[1,0]<stderr>:[07:43:02] ../src/operator/nn/./cudnn/./cudnn_algoreg-inl.h:[1,0]<stderr>:120: Running performance tests to find the best convolution algorithm, this can take a while... (set the environment variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
[1,0]<stdout>:DLL 2021-03-24 07:43:22.350445 - Epoch: 0 Iteration: 19  train.loss : 7.1160707712173465  train.ips : 396.43375862933976  train.lr : 0.0004936007640878701
[1,0]<stdout>:DLL 2021-03-24 07:43:27.017431 - Epoch: 0 Iteration: 39  train.loss : 7.1034677743911745  train.ips : 1748.5408471064911  train.lr : 0.0010131805157593122
[1,0]<stdout>:DLL 2021-03-24 07:43:31.665474 - Epoch: 0 Iteration: 59  train.loss : 7.085387134552002  train.ips : 1755.6720625808957  train.lr : 0.0015327602674307544
[1,0]<stdout>:DLL 2021-03-24 07:43:36.376168 - Epoch: 0 Iteration: 79  train.loss : 7.051189494132996  train.ips : 1732.288606178515  train.lr : 0.002052340019102197
[1,0]<stdout>:DLL 2021-03-24 07:43:41.040880 - Epoch: 0 Iteration: 99  train.loss : 7.0273662328720095  train.ips : 1749.3604844262006  train.lr : 0.002571919770773639
[1,0]<stdout>:DLL 2021-03-24 07:43:45.809392 - Epoch: 0 Iteration: 119  train.loss : 7.000608038902283  train.ips : 1711.3552677488967  train.lr : 0.0030914995224450813
[1,0]<stdout>:DLL 2021-03-24 07:43:50.608606 - Epoch: 0 Iteration: 139  train.loss : 6.96609468460083  train.ips : 1700.5316479155576  train.lr : 0.0036110792741165234
[1,0]<stdout>:DLL 2021-03-24 07:43:55.288886 - Epoch: 0 Iteration: 159  train.loss : 6.940982961654663  train.ips : 1743.575506725769  train.lr : 0.004130659025787966
[1,0]<stdout>:DLL 2021-03-24 07:43:59.984973 - Epoch: 0 Iteration: 179  train.loss : 6.917172503471375  train.ips : 1737.7140479540853  train.lr : 0.004650238777459409
[1,0]<stdout>:DLL 2021-03-24 07:44:04.747076 - Epoch: 0 Iteration: 199  train.loss : 6.893707418441773  train.ips : 1713.6071374694566  train.lr : 0.005169818529130851
[1,0]<stdout>:DLL 2021-03-24 07:44:09.422428 - Epoch: 0 Iteration: 219  train.loss : 6.876926112174988  train.ips : 1745.38353155098  train.lr : 0.005689398280802292
[1,0]<stdout>:DLL 2021-03-24 07:44:14.128075 - Epoch: 0 Iteration: 239  train.loss : 6.8568830490112305  train.ips : 1734.1632585620127  train.lr : 0.006208978032473735
[1,0]<stdout>:DLL 2021-03-24 07:44:18.746936 - Epoch: 0 Iteration: 259  train.loss : 6.847488880157471  train.ips : 1766.7605280161629  train.lr : 0.006728557784145177
[1,0]<stdout>:DLL 2021-03-24 07:44:23.413536 - Epoch: 0 Iteration: 279  train.loss : 6.827921938896179  train.ips : 1748.6550193963637  train.lr : 0.007248137535816619
[1,0]<stdout>:DLL 2021-03-24 07:44:28.094311 - Epoch: 0 Iteration: 299  train.loss : 6.804519271850586  train.ips : 1743.3676827998268  train.lr : 0.0077677172874880615

**Comment by VectorSL** [comment]
Thanks for your reply. 
I create a [topic](https://forums.developer.nvidia.com/t/which-mxnet-image-is-the-fastest/173576) in NVIDIA USER FORUM. 
I think PCIE and SXM does not have much difference in single card performance. 
![image](https://user-images.githubusercontent.com/16863563/112804130-9abd4c00-90a6-11eb-8483-ac15e92f8f38.png)

**Comment by VectorSL** [comment]
> Hi
> 
> The results from the page you linked are from MLPerf training model. This is a different model with different purpose.
> MLPerf focus to train the model to specific accuracy as fast as possible.
> Deep Learning Examples models are balanced between performance and final accuracy.
> 
> For example we don't recommend to train standard model with BS>256 due to final accuracy drop with large global batch size.
> However I didn't ever check how it behaves in slingle GPU case.
> Expected performance for MxNet model is approx ~2000 img/s with BS 256 and a single GPU on DGX A100
> 
> I see you are using different hardware configuration and PCIE version of A100 is slower than SXM used in DGX due to lower TDP
Thanks for your reply. 
I create a [topic](https://forums.developer.nvidia.com/t/which-mxnet-image-is-the-fastest/173576) in NVIDIA USER FORUM. 
I think PCIE and SXM does not have much difference in single card performance. 
![image](https://user-images.githubusercontent.com/16863563/112804130-9abd4c00-90a6-11eb-8483-ac15e92f8f38.png)

**Comment by VectorSL** [comment]
> Hi
> 
> The results from the page you linked are from MLPerf training model. This is a different model with different purpose.
> MLPerf focus to train the model to specific accuracy as fast as possible.
> Deep Learning Examples models are balanced between performance and final accuracy.
> 
> For example we don't recommend to train standard model with BS>256 due to final accuracy drop with large global batch size.
> However I didn't ever check how it behaves in slingle GPU case.
> Expected performance for MxNet model is approx ~2000 img/s with BS 256 and a single GPU on DGX A100
> 
> I see you are using different hardware configuration and PCIE version of A100 is slower than SXM used in DGX due to lower TDP



> Hi
> 
> The results from the page you linked are from MLPerf training model. This is a different model with different purpose.
> MLPerf focus to train the model to specific accuracy as fast as possible.
> Deep Learning Examples models are balanced between performance and final accuracy.
> 
> For example we don't recommend to train standard model with BS>256 due to final accuracy drop with large global batch size.
> However I didn't ever check how it behaves in slingle GPU case.
> Expected performance for MxNet model is approx ~2000 img/s with BS 256 and a single GPU on DGX A100
> 
> I see you are using different hardware configuration and PCIE version of A100 is slower than SXM used in DGX due to lower TDP

Thanks for your reply. 
I create a [topic](https://forums.developer.nvidia.com/t/which-mxnet-image-is-the-fastest/173576) in NVIDIA USER FORUM. 
I think PCIE and SXM does not have much difference in single card performance. 
![image](https://user-images.githubusercontent.com/16863563/112804130-9abd4c00-90a6-11eb-8483-ac15e92f8f38.png)

**Comment by hXl3s** [comment]
Hi @VectorSL,

after debugging, I manage to figure out the difference. Running docker with `--security-opt seccomp=unconfined` or `--privileged` could give additional boost. However still, the difference in TDP between PCIe and SXM card could lead to performance difference

**Comment by VectorSL** [comment]
@hXl3s Thanks a lot. I will try these docker option.  I'm short of a SXM card，so I can't make any further attempts at the card mod of TDP. Thanks again.

**Comment by xx205** [comment]
I recently tested the MXNet ResNet50v1.5 training performance with the single NVIDIA A100 SXM4 instance on Tencent Cloud.
The performance was better than that shown in [NVIDIA Data Center Deep Learning Product Performance].(https://developer.nvidia.com/deep-learning-performance-training-inference)

|Batch Size | images/s  | 
|------------| ------------- | 
|128 | 2812 |
|256 | 2964 |
|512 | 3078 |

Maybe you need to upgrade your driver version to 460.73.01 or 465.27.
